src/library/filescanner.c

```diff
@@ -1719,6 +1733,99 @@ filescanner_fullrescan()
   return 0;
 }
 
+static int
+filescanner_sync_metadata(const char *virtual_path, const uint32_t *id, uint32_t rating)
+{
+  int ret;
+  unsigned long flags;
+  char id_path[PATH_MAX] = { 0 };
+  char inotify_path[PATH_MAX] = { 0 };
+  struct watch_info wi = { 0 };
+  char *path;
+  int id_kind = 0;
+
+  if (virtual_path == NULL && id == NULL)
+    {
+      DPRINTF(E_LOG, L_SCAN, "Bug! Invalid params %s:%d\n", __FILE__, __LINE__);
+      return -1;
+    }
+
+  if (virtual_path)
+    {
+      path = (char*)virtual_path_to_path(virtual_path);
+      if (!path)
+	{
+	  DPRINTF(E_INFO, L_SCAN, "Ignoring non local media %s requested for metadata sync\n", virtual_path);
+	  return -1;
+	}
+    }
+  else
+    {
+      // Determine if this is local media
+      id_kind = db_file_data_kind_byid(*id);
+      if (id_kind != DATA_KIND_FILE)
+	{
+	  DPRINTF(E_INFO, L_SCAN, "Ignoring non local media id=%d (%s) requested for metadata sync\n", *id,db_data_kind_label(id_kind));
+	  return -1;
+	}
+
+      path = db_file_path_byid(*id);
+      if (!path)
+        {
+	  DPRINTF(E_LOG, L_SCAN, "No known path for local media id=%d\n", *id);
+	  return -1;
+	}
+      strcpy(id_path, path);
+      free(path);
+
+      path = id_path;
+    }
+
+  // Inotify watches dir paths
```

Comment:
in my `inotify` table, all paths are directories - therefore i need to get the dir path, since the inotify backend "watches" directory paths only.  does that clear it up?